src/components/ContactsAdd.js

```diff
@@ -1,31 +1,164 @@
-import { useState } from "react"
+import { useEffect, useRef, useState } from "react"
+// import {useForm} from "react-hook-form"
 import { useNavigate } from "react-router-dom";
 
+const initialState = {
+  firstName: "",
+  lastName: "",
+  street: "",
+  city: "",
+  email: "",
+  twitter: "",
+  linkedIn: "",
+}
+
+
 function ContactsAdd(props) {
 
-  // setContacts and contacts must be passed as props
-  // to this component so new contacts can be added to the
-  // state
   const { setContacts, contacts } = props
+  const [formState, setFormState] = useState(initialState)
+  //calling this function will allow us to change the current page
+  const navigate = useNavigate()
 
-  //TODO: Implement controlled form
   //send POST to json server on form submit
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    setContacts([...contacts, formState])
+
+    //POST request
+    const opts = {
+      method: 'POST',
+      //stackoverflow says it has to be "application/json"
+      headers: { 'content-type': 'application/json' },
+      body: JSON.stringify({
+        "firstName": formState.firstName,
+        "lastName": formState.lastName,
+        "street": formState.street,
+        "city": formState.city,
+        "email": formState.email,
+        "twitter": formState.twitter,
+        "linkedIn": formState.linkedIn,
+      })
+    }
+    //Pass the URL we want to pass TO
+    fetch("http://localhost:4000/contacts", opts)
+      .then(res => res.json())
+      .then(data => {
+        formState
+        // console.log("posted contacts:", contacts)
+        console.log("posted contacts formState:", formState)
+      })
+    //reset the form. Timer to allow for post request to complete
+    setTimeout(() => {
+      setFormState(initialState);
+    }, 500)
+
+    //GET request to get teh data back with an ID
+    fetch("http://localhost:4000/contacts")
```

Comment:
The post request already sends back the newly created entry in the database so you can use that response instead of creating a get request.